File: library/subsystem/libsubsystem/cppsubsystem/cqsubsystem.cpp
#include "cqsubsystem.hh"
#include "cqsubsystemarche.h"

cq_member(cqSubsystemManager) {
};

cqSubsystemManager::cqSubsystemManager() {
}

cqSubsystemManagerRef cqSubsystemManager::get() {
    return cqStaticObject<cqSubsystemManager>();
}

void cqSubsystemManager::startSubsystem() {
    cq_subsystem_start();
}

void cqSubsystemManager::stopSubsystem() {
    cq_subsystem_stop();
}

File: library/subsystem/IOSSubsystem/CQSubsystem.m
#import "CQSubsystem.h"
#import "cqsubsystemarche.h"

bool cq_subsystem_start(const char *name) {
    return [CQSubsystemManager.sharedObject startSubsystem:@(name)];
}

void cq_subsystem_stop(const char *name) {
    [CQSubsystemManager.sharedObject stopSubsystem:@(name)];
}
